src/pytorch_lightning/strategies/ipu.py

```diff
@@ -272,6 +269,16 @@ def to_tensor(x):
         args = apply_to_collection(args, dtype=(int, float), function=to_tensor)
         return args
 
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
+        # This override is necessary because the cast must occur before the data
+        # is moved to the device to prevent wasteful host->device copies.
+        if self.precision_plugin.precision in (PrecisionType.MIXED, PrecisionType.HALF):
+            batch = apply_to_collection(batch, (FloatTensor, torch.cuda.FloatTensor), function=Tensor.half)
```

Comment:
```suggestion
            batch = apply_to_collection(batch, Tensor, function=Tensor.half)
```
if we need to move all kinds of tensors for IPUs

Reply:
Is this the change that caused https://github.com/Lightning-AI/lightning/pull/13983?

Reply:
that's part of it, but before e.g. DoubleTensors, BfloatTensors etc. wouldn't have been cast as well.